### ISSUE TITLE
Trouble cross compiling: sybfront.h not found, and therefore, freetds is missing

### ISSUE THREAD

**Issue by aharpervc** [rails-sqlserver/tiny_tds#331]
**Problem:**

I'm trying to run `bundle exec gem:windows` to build the gem for Windows, from Windows. But I'm having a problem during the build process:

```
(snip)
Running 'install' for freetds 1.00.21... OK
Activating freetds 1.00.21 (from /c/users/aharper/workspace/tiny_tds/ports/i686-w64-mingw32)...
checking for sybfront.h... no
-----
freetds is missing.
Do you have FreeTDS 0.95.80 or higher installed?
-----
*** ../../../../ext/tiny_tds/extconf.rb failed ***
(snip)
```

**Environment:**

* master branch, up to date
* Host environment is Windows 10 x64, using Hyper-V to boot docker machine, with the repo folder mounted inside the machine
* I changed the Rakefile to only attempt to build for ruby 2.4.0

**Situation:**

Docker is booting and running `bundle && rake cross native gem RUBY_CC_VERSION=2.4.0 CFLAGS="-Wall"` inside the docker machine, and this process fails with the above error.

mkmf.log says:

```
"i686-w64-mingw32-gcc -E -I/usr/local/rake-compiler/ruby/i686-w64-mingw32/ruby-2.4.0/include/ruby-2.4.0/i386-mingw32 -I/usr/local/rake-compiler/ruby/i686-w64-mingw32/ruby-2.4.0/include/ruby-2.4.0/ruby/backward -I/usr/local/rake-compiler/ruby/i686-w64-mingw32/ruby-2.4.0/include/ruby-2.4.0 -I../../../../ext/tiny_tds -I/c/users/aharper/workspace/tiny_tds/ports/i686-w64-mingw32/include -D_WIN32_WINNT=0x0501 -D__MINGW_USE_VC2005_COMPAT -D_FILE_OFFSET_BITS=64  -s -O1 -fno-omit-frame-pointer -fno-fast-math -Wall  conftest.c -o conftest.i"
cc1: error: \c\users\aharper\workspace\tiny_tds\ports\i686-w64-mingw32\include: Invalid argument
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: #include <sybfront.h>
/* end */
```

I can confirm that `sybfront.h` **does** exist (along with `sybdb.h` and the other header files) in `ports/i686-w64-mingw32/include`. **It appears that the problem may be the directory slashes going the wrong way... somewhere.** Even though the include argument is `-I/c/users/aharper/workspace/tiny_tds/ports/i686-w64-mingw32/include`, which should be fine.

I grep'd for part of that path to see where it might be coming out with backslashes in but everywhere looks correct with forward slashes. Does this seem like the actual problem? Where could this be happening? 

**Full trace:**

```
$ bundle exec rake gem:windows
rake-compiler-dock bash -c "    bundle && rake cross native gem RUBY_CC_VERSION=2.4.0 CFLAGS=\"-Wall\"\n"
The git source `git://github.com/rake-compiler/rake-compiler-dock.git` uses the `git` protocol, which transmits data without encryption. Disable this warning with `bundle config git.allow_insecure true`, or switch to the `https` protocol to keep your data secure.
Fetching git://github.com/rake-compiler/rake-compiler-dock.git
Fetching gem metadata from https://rubygems.org/.............
Fetching version metadata from https://rubygems.org/.
Resolving dependencies...
Installing rake 10.5.0
Installing byebug 9.0.6 with native extensions
Installing connection_pool 2.2.1
Using mini_portile2 2.1.0
Using minitest 5.10.1
Using rake-compiler-dock 0.6.0 from git://github.com/rake-compiler/rake-compiler-dock.git (at master@da5ecc3)
Using bundler 1.13.7
Installing rake-compiler 0.9.5
Using tiny_tds 1.1.0 from source at `.`
Bundle complete! 7 Gemfile dependencies, 9 gems now installed.
Use `bundle show [gemname]` to see where a bundled gem is installed.
mkdir -p tmp/x86-mingw32/tiny_tds/2.4.0
cp /home/rvm/.rake-compiler/ruby/i686-w64-mingw32/ruby-2.4.0/lib/ruby/2.4.0/mkmf.rb tmp/x86-mingw32/tiny_tds/2.4.0/mkmf.rb
cd tmp/x86-mingw32/tiny_tds/2.4.0
/usr/local/rvm/rubies/ruby-2.4.0/bin/ruby -I. ../../../../ext/tiny_tds/extconf.rb --disable-lookup --enable-cross-build
../../../../ext/tiny_tds/extconf.rb:14: warning: already initialized constant OpenSSL::SSL::VERIFY_PEER
checking for sybfront.h... no
Downloading openssl-1.0.2j.tar.gz (100%)
Extracting openssl-1.0.2j.tar.gz into tmp/i686-w64-mingw32/ports/openssl/1.0.2j... OK
Running 'configure' for openssl 1.0.2j... OK
Running 'compile' for openssl 1.0.2j... OK
Running 'mkdef-libeay32' for openssl 1.0.2j... OK
Running 'mkdef-ssleay32' for openssl 1.0.2j... OK
Running 'gcc-libeay32-1.0.2j-i686-w64-mingw32-compile' for openssl 1.0.2j... OK
Running 'dlltool-libeay32-1.0.2j-i686-w64-mingw32-exp' for openssl 1.0.2j... OK
Running 'gcc-libeay32-1.0.2j-i686-w64-mingw32-dll' for openssl 1.0.2j... OK
Running 'dlltool-libeay32-1.0.2j-i686-w64-mingw32-outputlib' for openssl 1.0.2j... OK
Running 'gcc-libeay32-1.0.2j-i686-w64-mingw32-link' for openssl 1.0.2j... OK
Running 'gcc-ssleay32-1.0.2j-i686-w64-mingw32-compile' for openssl 1.0.2j... OK
Running 'dlltool-ssleay32-1.0.2j-i686-w64-mingw32-exp' for openssl 1.0.2j... OK
Running 'gcc-ssleay32-1.0.2j-i686-w64-mingw32-dll' for openssl 1.0.2j... OK
Running 'dlltool-ssleay32-1.0.2j-i686-w64-mingw32-outputlib' for openssl 1.0.2j... OK
Running 'gcc-ssleay32-1.0.2j-i686-w64-mingw32-link' for openssl 1.0.2j... OK
Running 'install' for openssl 1.0.2j... OK
Activating openssl 1.0.2j (from /c/users/aharper/workspace/tiny_tds/ports/i686-w64-mingw32)...
Downloading libiconv-1.14.tar.gz (100%)
Extracting libiconv-1.14.tar.gz into tmp/i686-w64-mingw32/ports/libiconv/1.14... OK
Running 'configure' for libiconv 1.14... OK
Running 'compile' for libiconv 1.14... OK
Running 'install' for libiconv 1.14... OK
Activating libiconv 1.14 (from /c/users/aharper/workspace/tiny_tds/ports/i686-w64-mingw32)...
Downloading freetds-1.00.21.tar.bz2 (100%)
Extracting freetds-1.00.21.tar.bz2 into tmp/i686-w64-mingw32/ports/freetds/1.00.21... OK
Running git apply with /c/users/aharper/workspace/tiny_tds/ports/patches/freetds/1.00.21/0001-mingw_missing_inet_pton.diff... OK
Running 'configure' for freetds 1.00.21... OK
Running 'compile' for freetds 1.00.21... OK
Running 'install' for freetds 1.00.21... OK
Activating freetds 1.00.21 (from /c/users/aharper/workspace/tiny_tds/ports/i686-w64-mingw32)...
checking for sybfront.h... no
-----
freetds is missing.
Do you have FreeTDS 0.95.80 or higher installed?
-----
*** ../../../../ext/tiny_tds/extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
        --with-opt-dir
        --without-opt-dir
        --with-opt-include
        --without-opt-include=${opt-dir}/include
        --with-opt-lib
        --without-opt-lib=${opt-dir}/lib
        --with-make-prog
        --without-make-prog
        --srcdir=../../../../ext/tiny_tds
        --curdir
        --ruby=/usr/local/rake-compiler/ruby/i686-w64-mingw32/ruby-2.4.0/bin/$(RUBY_BASE_NAME)
        --help
        --enable-lookup
        --disable-lookup
        --with-freetds-dir
        --without-freetds-dir
        --with-freetds-include
        --without-freetds-include=${freetds-dir}/include
        --with-freetds-lib
        --without-freetds-lib=${freetds-dir}/lib
        --enable-system-freetds
        --disable-system-freetds
        --enable-system-iconv
        --disable-system-iconv
        --enable-system-openssl
        --disable-system-openssl
        --enable-gnutls
        --disable-gnutls
        --enable-openssl
        --disable-openssl
        --enable-cross-build
        --with-freetds-dir
        --without-freetds-dir
        --with-freetds-include
        --without-freetds-include=${freetds-dir}/include
        --with-freetds-lib
        --without-freetds-lib=${freetds-dir}/lib
rake aborted!
Command failed with status (1): [/usr/local/rvm/rubies/ruby-2.4.0/bin/ruby ...]
/usr/local/rvm/gems/ruby-2.4.0/gems/rake-compiler-0.9.5/lib/rake/extensiontask.rb:191:in `block (2 levels) in define_compile_tasks'
/usr/local/rvm/gems/ruby-2.4.0/gems/rake-compiler-0.9.5/lib/rake/extensiontask.rb:188:in `block in define_compile_tasks'
/usr/local/rvm/gems/ruby-2.4.0/bin/ruby_executable_hooks:15:in `eval'
/usr/local/rvm/gems/ruby-2.4.0/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => native => native:x86-mingw32 => native:tiny_tds:x86-mingw32 => tmp/x86-mingw32/stage/lib/tiny_tds/tiny_tds.so => copy:tiny_tds:x86-mingw32:2.4.0 => tmp/x86-mingw32/tiny_tds/2.4.0/tiny_tds.so => tmp/x86-mingw32/tiny_tds/2.4.0/Makefile
(See full trace by running task with --trace)
rake aborted!
Command failed with status (1): [docker run -v /c/users/aharper/workspace/tiny_tds:/c/users/aharper/workspace/tiny_tds -e UID\=1000 -e GID\=1000 -e USER\=aharper -e GROUP\=dummygroup -e ftp_proxy\= -e http_proxy\= -e https_proxy\= -e RCD_HOST_RUBY_PLATFORM\=i386-mingw32 -e RCD_HOST_RUBY_VERSION\=2.3.3 -e RCD_IMAGE\=larskanis/rake-compiler-dock:0.6.0 -w /c/users/aharper/workspace/tiny_tds --rm -i -t larskanis/rake-compiler-dock:0.6.0 runas sigfw bash -c \ \ \ \ bundle\ \&\&\ rake\ cross\ native\ gem\ RUBY_CC_VERSION\=2.4.0\ CFLAGS\=\"-Wall\"'
']
C:/users/aharper/workspace/tiny_tds/Rakefile:91:in `block in <top (required)>'
Tasks: TOP => gem:windows
(See full trace by running task with --trace)
```

**Comment by aharpervc** [comment]
@larskanis @metaskills any ideas?

**Comment by aharpervc** [comment]
Naturally the instant I post this, I try something and it works.

`tmp/x86-mingw32/tiny_tds/2.4.0/fake.rb` defines `ALT_SEPARATOR` as `"\\"`. I changed it to "/" and the compilation succeeds for x86, failing again for x64. I changed that fake.rb as well, and reran the rake command, and everything appeared to succeed.

Now to find out who's setting that and why and if it should be set based on the compile host environment (linux) and not the target environment (Windows).
